services/provider/server/server.go

```diff
@@ -747,12 +729,20 @@ func (s *OCSProviderServer) ReportStatus(ctx context.Context, req *pb.ReportStat
 		return nil, status.Errorf(codes.Internal, "Failed to update lastHeartbeat payload in the storageConsumer resource: %v", err)
 	}
 
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, status.Errorf(codes.NotFound, "Failed to get the storageConsumer to set storage quota: %v", err)
+	}
+
+	clusterResourceQuotaSpec := s.getClusterResourceQuotaSpec(storageConsumer)
 	channelName, err := s.getOCSSubscriptionChannel(ctx)
 	if err != nil {
 		return nil, status.Errorf(codes.Internal, "Failed to construct status response: %v", err)
 	}
 
-	return &pb.ReportStatusResponse{DesiredClientOperatorChannel: channelName}, nil
+	desiredProviderConfigHash := util.GetMD5Hash(fmt.Sprintf("%v%s", clusterResourceQuotaSpec, channelName))
```

Comment:
If we also include the hash for mon-endpoints, we could remove the [GetStorageConfig](https://github.com/red-hat-storage/ocs-client-operator/blob/main/service/status-report/main.go#L121-L143) call from status-reporter and update this endpoint in storageClient controller along with creating/Updating [clusterResourceQuota](https://github.com/red-hat-storage/ocs-client-operator/blob/main/internal/controller/storageclient_controller.go#L202-L214).

This will allow us to call only`ReportStatus` from status-reporter.

WDYT @nb-ohad @leelavg ?

Reply:
I kinda saw the fine print of the implementation in https://docs.google.com/document/d/1zJr_IWH9YeX1LukYjfh68duzyIfb7bdn3t8rI7UbQwQ/edit?disco=AAABSUNF_fE, however I believe code here isn't matching the expectations.

Let me take this opportunity to have a terser explanation of conversations that we had:

t0:
client tries to onboard onto provider, here we could send client identifiable static data (ex: clusterid, custername) to the provider and potentially remove the msg passing of this static data many times.

t1:
once the client is onboarded, the heartbeat kicks in, the original intention of heartbeat is whether client & provider is discoverable to each other or not and send any data that changes over time (mostly versions)

as a result, provider could maintain (or generate on the fly) the dynamic info that it wants to send to client (suppose only the channel name, quota, mons), however even the dynamic info is static for a longer period of time and so we want to take hash of all the dynamic info and send as heartbeat response to client.

now heartbeat just updates the hash onto storageclient status (preferably) and corresponding controller would refetch data if the status is updated.

I do not want to expand more, pls excuse if above doesn't strike anything, we can discuss here/meet and close this at the earliest.

-------

> If we also include the hash for mon-endpoints, we could remove the [GetStorageConfig](https://github.com/red-hat-storage/ocs-client-operator/blob/main/service/status-report/main.go#L121-L143) call
- I see we are sending more info than mons but client is only consuming mons and so there is not need to remove the RPC

> This will allow us to call onlyReportStatus from status-reporter.
- I get this, I believe this stemmed from having ceph-csi-config as the sink to the data received from more than one RPC call
- due to ceph-csi-op which demarcates the responsibilities of what data goes where and I would like the current RPCs to be unchanged, give it some more time on how the integration helps the overall flow.


Nevertheless, open to be corrected.

Reply:
PR is updated now, Got reviews so soon which haven't expected to happen :-). @leelavg yes doing the changes as you explained the above.

Reply:
got LGTM in comments and so resolving the conv.